doc/stdenv/stdenv.xml

```diff
@@ -2111,16 +2111,106 @@ postInstall = ''
       </para>
      </listitem>
     </varlistentry>
+
     <varlistentry>
      <term>
-      cmake
+      <anchor xml:id="cmake"/>
+       cmake
      </term>
      <listitem>
       <para>
-       Overrides the default configure phase to run the CMake command. By default, we use the Make generator of CMake. In addition, dependencies are added automatically to CMAKE_PREFIX_PATH so that packages are correctly detected by CMake. Some additional flags are passed in to give similar behavior to configure-based packages. You can disable this hook’s behavior by setting configurePhase to a custom value, or by setting dontUseCmakeConfigure. cmakeFlags controls flags passed only to CMake. By default, parallel building is enabled as CMake supports parallel building almost everywhere. When Ninja is also in use, CMake will detect that and use the ninja generator.
+       Overrides the default configure phase to run the CMake command. By default, we use the Make generator of CMake, however, if ninja is present, then the ninja generator will be used instead. In addition, dependencies are added automatically to CMAKE_PREFIX_PATH so that packages are correctly detected by CMake. By default, the cmake configurePhase can be thought of as:
+       <programlisting>
+mkdir build
+cd build
+cmake ..</programlisting>
       </para>
+      <variablelist>
+       <title>Variables controlling CMake</title>
+       <varlistentry>
+        <term>
+         <varname>cmakeFlags</varname>
+        </term>
+        <listitem>
+         <para>
+          Flags passed to cmake. <literal>cmakeFlags = "-DBUILD_SHARED_LIB=ON -DENABLE_LTO=ON";</literal>
+         </para>
+        </listitem>
+       </varlistentry>
+       <varlistentry>
+        <term>
+         <varname>cmakeFlagsArray</varname>
+        </term>
+        <listitem>
+         <para>
+          Flag array passed to cmake. <literal>cmakeFlagsArray = [ "-DBUILD_SHARED_LIB=ON" "-DENABLE_LTO=ON" ];</literal>
```

Comment:
This should not really be used in Nix since the stringification for bash will destroy any boundaries again. It is used in `preConfigure` when one needs to pass a flag containing whitespace.